NorthstarDLL/core/structs.h

```diff
@@ -25,38 +25,24 @@ OFFSET_STRUCT(Name)
 
 #define OFFSET_STRUCT(name) union name
 #define STRUCT_SIZE(size) char __size[size];
-#define STRUCT_FIELD_OFFSET(offset, signature)                                                                                             \
-	struct                                                                                                                                 \
-	{                                                                                                                                      \
-		char CONCAT2(pad, __LINE__)[offset];                                                                                               \
-		signature;                                                                                                                         \
-	};
+#define STRUCT_FIELD_OFFSET(offset, signature) \
+	char CONCAT2(pad, __LINE__)[offset]; \
+	signature;
 
 // Special case for a 0-offset field
 #define STRUCT_FIELD_NOOFFSET(offset, signature) signature;
 
-// Based on: https://stackoverflow.com/questions/11632219/c-preprocessor-macro-specialisation-based-on-an-argument
-// Yes, this is hacky, but it works quite well actually
-// This basically makes sure that when the offset is 0x0, no padding field gets generated
-#define OFFSET_0x0 ()
+#define FX(f,x) f x
 
-#define IIF(c) CONCAT2(IIF_, c)
-#define IIF_0(t, ...) __VA_ARGS__
-#define IIF_1(t, ...) t
+#define TEST_0 ,
```

Comment:
ignore the build failure, despite my writing I forgot that there also needs to be a `TEST_0x0`, because MSVC does not convert the integer constant `0x0` to `0` before.

Reply:
Tested, with this it builds on MSVC again.
Holding off on making a commit because of the formatting issue.